ui/app/services/nomad-actions.js

```diff
@@ -0,0 +1,310 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+// @ts-check
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { base64DecodeString } from '../utils/encode';
+import config from 'nomad-ui/config/environment';
+
+export default class NomadActionsService extends Service {
```

Comment:
This is the global service that handles the majority of what the user interacts with when they run an action:
- It maintains the queue of running/ran action instances
- It handles opening and reacting to events over WebSocket
- It handles checks for number of running/stopped actions (so we can show count around the app)
- It handles interactivity like clearing/stopping an action
- It determines if the sidebar/flyout should be opened or shut